query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/TrinoBasedRequestHandler.java

```diff
@@ -130,8 +130,7 @@ public Observable<Row> handleRequest(QueryRequest request, ExecutionContext exec
 
       return executeQuery(sql.getKey(), sql.getValue());
     } catch (Throwable t) {
-      String truncatedMessage = (t.getMessage() == null) ? null : t.getMessage().substring(0, 2048);
-      return Observable.error(new Throwable(truncatedMessage));
+      return Observable.error(new Throwable(truncateMessage(t.getMessage(), 2048)));
```

Comment:
Can we use existing library function?
```
StringUtils.truncate(t.getMessage(), 2048)
```
Also define maxExceptionLength as private static varibale.

Reply:
StringUtils.truncate takes care of null check, out of index array, etc.

Reply:
Wanted to avoid 3p lib dependency